### PR TITLE
Dyanmic app querying

### PR DESCRIPTION
**Pull request by DrewWeth**
Able to querying by filtering a client's application. You can only do strings right now. I'm not sure how to represent a JSON object in GoLang. I.e. I can't make a map of different types: strings/ints/floats

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,32 +1,69 @@
...
+
+	router.HandleFunc("/{client}/{application}", application.QueryApplicationHandler).Methods("POST")
+	router.HandleFunc("/{client}/{application}/new", application.CreateApplicationHandler).Methods("POST")
+
 	defer client.Close()
 
 	n := negroni.Classic()
 	n.UseHandler(router)
-	n.Run(":3001")
+	n.Run(":3000")
```
Comment (Felttrip) [inline main.go]: We should keep this on 3001 unless we really need it to be on 3000

File: main.go [hunk 2]
```diff
@@ -1,32 +1,69 @@
...
-	})
-
 	// TODO: Use command line flag credentials.
 	client, err := db.NewClient("localhost:28015")
 	if err != nil {
 		log.Fatal("Couldn't initialize database: ", err.Error())
 	}
+	db.CreateEaseDb(client)
+	db.CreateUserTable(client)
+	db.CreateDbTable(client)
```
Comment (Felttrip) [inline main.go]: Should we be creating the db and tables every time we start the server?

File: main.go (hunk 1 above)
Comment (DrewWeth) [inline main.go]: It was either change this line or change the vagrant file and reload it, this seemed easier. The vagrant file is only port-forwarded for 3000

File: main.go (hunk 2 above)
Comment (DrewWeth) [inline main.go]: The dbs and tables aren't overwritten by default, you just get an error and it's logged. I could see adding a check being a good val-cap

**Pull request merged**